kaggle_environments/envs/llm_20_questions/llm_20_questions.py

```diff
@@ -171,42 +174,47 @@ def interpreter(state, env):
     end_early = (active1 and active1.status) in (TIMEOUT, ERROR) or (active2 and active2.status in (TIMEOUT, ERROR))
     one_guessed = False
     two_guessed = False
+    one_bad_response = False
+    two_bad_response = False
 
-    if active1 is not None:
-        guessed = False
-        if active1.observation.role == GUESSER:
-            guessed = guesser_action(active1, inactive1, step)
-            one_guessed = guessed
-        else:
-            answerer_action(active1, inactive1)
+    if active1 is None or active2 is None:
+        raise ValueError
 
-        if active1.status in (TIMEOUT, ERROR):
-            end_game(active1, inactive1, 0, active1.status, DONE)
-        elif end_early:
-            end_game(active1, inactive1, 0, DONE, DONE)
-        else:
-            increment_turn(active1, inactive1, step, guessed)
-    
-    if active2 is not None:
-        guessed = False
-        if active2.observation.role == GUESSER:
-            guessed = guesser_action(active2, inactive2, step)
-            two_guessed = guessed
-        else:
-            answerer_action(active2, inactive2)
+    if active1.observation.role == GUESSER:
+        one_guessed = guesser_action(active1, inactive1, step)
+    else:
+        one_bad_response = answerer_action(active1, inactive1)
 
-        if active2.status in (TIMEOUT, ERROR):
-            end_game(active2, inactive2, 0, active2.status, DONE)
-        elif end_early:
-            end_game(active2, inactive2, 0, DONE, DONE)
-        else:
-            increment_turn(active2, inactive2, step, guessed)
+    if active2.observation.role == GUESSER:
+        two_guessed = guesser_action(active2, inactive2, step)
+    else:
+        two_bad_response = answerer_action(active2, inactive2)
+
+    if active1.status in (TIMEOUT, ERROR) or one_bad_response:
+        end_game(active1, -1, active1.status)
+        end_game(inactive1, 1, DONE)
+    elif end_early or two_bad_response:
+        end_game(active1, 1, DONE)
+        end_game(inactive1, 1, DONE)
+    else:
+        increment_turn(active1, inactive1, step, one_guessed)
+    
+    if active2.status in (TIMEOUT, ERROR) or two_bad_response:
+        end_game(active2, -1, active2.status)
+        end_game(inactive2, 1, DONE)
+    elif end_early or one_bad_response:
+        end_game(active2, 1, DONE)
+        end_game(inactive2, 1, DONE)
+    else:
+        increment_turn(active2, inactive2, step, two_guessed)
```

Comment:
Decent bit of mostly duplicated code that we could maybe unify in helper(s)?

Reply:
yeah I've been meaning to do that, I'll do that in a follow up